GDJS/Runtime/ResourceLoader.ts

```diff
@@ -553,4 +565,79 @@ namespace gdjs {
       return this._model3DManager;
     }
   }
+
+  type PromiseError<T> = { item: T; reason: any };
+
+  type PromisePoolOutput<T, U> = {
+    results: Array<U>;
+    errors: Array<PromiseError<T>>;
+  };
+
+  const promisePool = <T, U>(
+    items: Array<T>,
+    maxConcurrency: number,
+    asyncFunction: (item: T) => Promise<U>
+  ): Promise<PromisePoolOutput<T, U>> => {
+    const results: Array<U> = [];
+    const errors: Array<PromiseError<T>> = [];
+    let activePromises = 0;
+    let index = 0;
+
+    return new Promise((resolve, reject) => {
+      const executeNext = () => {
+        if (items.length === 0) {
+          resolve({ results, errors });
+          return;
+        }
+        while (activePromises < maxConcurrency && index < items.length) {
+          const item = items[index++];
+          activePromises++;
+
+          asyncFunction(item)
+            .then((result) => results.push(result))
+            .catch((reason) => errors.push({ item, reason }))
+            .finally(() => {
+              activePromises--;
+              if (index === items.length && activePromises === 0) {
+                resolve({ results, errors });
+              } else {
+                executeNext();
+              }
+            });
+        }
+      };
+
+      executeNext();
+    });
+  };
+
+  const promisePoolAndRetry = async <T, U>(
+    items: Array<T>,
+    maxConcurrency: number,
+    maxAttempt: number,
+    asyncFunction: (item: T) => Promise<U>
+  ): Promise<PromisePoolOutput<T, U>> => {
```

Comment:
Let's use a verb here too:
```suggestion
  const processAndRetryIfNeededWithPromisePool = async <T, U>(
    items: Array<T>,
    options: {maxConcurrency: number, maxAttempt: number},
    asyncFunction: (item: T) => Promise<U>
  ): Promise<PromisePoolOutput<T, U>> => {
```